lib/graphql/query/validation_pipeline.rb

```diff
@@ -72,12 +74,14 @@ def ensure_has_validated
         elsif @operation_name_error
           @validation_errors << @operation_name_error
         else
-          validation_result = @schema.static_validator.validate(@query, validate: @validate, timeout: @schema.validate_timeout, max_errors: @schema.validate_max_errors)
-          @validation_errors.concat(validation_result[:errors])
-          @internal_representation = validation_result[:irep]
+          if !@skip_static_validation
+            validation_result = @schema.static_validator.validate(@query, validate: @validate, timeout: @schema.validate_timeout, max_errors: @schema.validate_max_errors)
```

Comment:
Instead of skipping this entire block, can we change the `validate` option to take `@skip_static_validation` into account? I'm thinking it would be less error prone if @internal_representation was always set. Also, it looks like it'll continue to validate variables below?

Something like:

```ruby
should_validate = @skip_static_validation ? false : @validate
validation_result = @schema.static_validator.validate(@query, validate: should_validate, timeout: @schema.validate_timeout, max_errors: @schema.validate_max_errors)
```

Reply:
yep, made the changes. It makes sense that input variables need to be validated for every request.